ui/puzzle/src/ctrl.ts

```diff
@@ -88,6 +101,7 @@ export default function (opts: PuzzleOpts, redraw: Redraw): Controller {
     vm.initialPath = initialPath;
     vm.initialNode = tree.nodeAtPath(initialPath);
     vm.pov = vm.initialNode.ply % 2 == 1 ? 'black' : 'white';
+    data.player = { color: vm.pov };
```

Comment:
This feels very hacky to me, but not sure how to do it better. I'm just sticking the color into `PuzzleData` where it exists in `RoundData`.

Reply:
Because you're passing the actual puzzle controller to the keyboardMove instance with `makeKeyboardMove(this, ...)`.

I was thinking of instanciating a new structure instead, like this:

```typescript
makeKeyboardMove({
  data: this.data,
  sendMove: ...,
  ...,
  redraw: this.redraw
})
```

This way keyboardMove can precisely declare what it needs, regardless of how the controllers instanciating it look like.

Reply:
Got it, this makes a lot of sense. I didn't consider doing it on the fly like this. I'll make some changes 👍